java/arcs/core/host/Resurrector.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.host
+
+import arcs.core.storage.StorageKey
+
+typealias ResurrectorCallback = (String, List<StorageKey>) -> Unit
+
+/**
+ * A [Resurrector] is supplied by the embedder (e.g. [Service]) of an [ArcHost] as a mechanism for
+ * requesting, or cancelling resurrection, as well as processing calls for resurrection.
+ */
+interface Resurrector {
+    /**
+     * Issue a request to be resurrected by a resurrection service whenever the data identified by
+     * the provided [storageKeys] changes.
+     */
+    fun requestResurrection(arcId: String, storageKeys: List<StorageKey>) = Unit
+
+    /** Cancel resurrection registrations for the given [arcId]. */
+    fun cancelResurrection(arcId: String) = Unit
+
+    /**
+     * Registers a callback to be called by the resurrection service whenever the [affectedKeys]
+     * have been updated.
+     */
+    fun onResurrection(hostId: String, block: ResurrectorCallback) = Unit
```

Comment:
Similar to arcId, maybe having hostId be an actual type would be beneficial at some point. We've a bunch of different kinds of ids for different things, and we had that recent discussion about how particles have two different kinds of ids, etc. so concretizing that in types starts to feel worthwhile.

Reply:
For now it could be an `inline class` extending `String`.

Reply:
I'll do this in a followup PR replacing all arcId, hostId, and other String "types", with either concrete classes (ArcId), inline classes (e.g. hostId), or enums (if that's what they're being used for as a kind of poor man's enum)